### DISCUSSION TITLE
how to use ros wrapper in windows?

### DISCUSSION THREAD

**Discussion by anguoyoula** [microsoft/AirSim#4642] [Support Q&A]
present situation：

1. install airsim in windows properly (include unrealengine and vs) and i can use api to control the drone.
2. i installed ros in windows before, but i can`t intall the dependency such as ros-noetic-tf2-sensor-msgs with choco, so i give up.

what i want to do is:

1. run airsim and unrealengine on windows and publish the ros data to another linux or wsl , how can i change the network port or waht i need to do ?

Thank you very much！

**Comment by jonyMarino** [comment]
Hi @anguoyoula! Have you taken a look at [the documentation](https://microsoft.github.io/AirSim/airsim_ros_pkgs/#setting-up-the-build-environment-on-windows10-using-wsl1-or-wsl2)?

**Reply by anguoyoula** [reply]
yes, fllowing the document, i installed VcXsrv and config it. and i started them as fllows:

1. close all firewall
2. start vcxsrv 
3. start wsl  and export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0 , then startxfce4
4. open another terminal with the desktop in wsl and cd Airsim/ros
5. export WSL_HOST_IP=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}')
6. source devel/setup.bash and run roslaunch airsim_ros_pkgs airsim_node.launch output:=screen host:=$WSL_HOST_IP
7. as the same , source devel/setup.bash and run roslaunch airsim_ros_pkgs rviz.launch
8. go back to windows and start airsim-project(i had replaced the airsim setting file with the new setting in ros package)
9. start python scripts to control drone take off

but i dont see any data display in wsl windows?

if there are some errors i did?

thank you very much!

**Reply by jonyMarino** [reply]
Are you using wsl or wsl2? Also, `echo $WSL_HOST_IP` prints your host IP correctly?
